### PR TITLE
Fix #8600: Accounts Tab v2

### LINKED ISSUE #8600: Accounts Tab v2
### Description: 

Update Accounts tab to v2 designs

### PR DESCRIPTION
**Pull request by StephenHeaps**
## Summary of Changes
- Updates account tab to v2 designs.
    - Backup & Add Account moved to `...` button.
    - Accounts now show tokens with balance and their total fiat value
    - Accounts have view details, edit, and export options in `...` context menu
- Backup now available on all tabs in the `...` menu. Add account available in `...` menu on Accounts tab only.

This pull request fixes #8600

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`
- [x] New or updated UI has been tested across:
  - [x] Light & dark mode
  - [x] Different size classes (iPhone, landscape, iPad)
  - [x] Different dynamic type sizes

## Test Plan:
1. Verify accounts tab still showing all accounts, with total fiat values and the tokens with most balance displayed
2. Tapping the account card or tapping `...` -> `View Details` should open account assets / transactions view
3. Tapping `...` -> `Edit` should open modal for the account to edit the account's name
4. Tapping `...` -> `Export` should open private key flow for the account
5. Verify Backup Wallet still opens/works as expected (now accessed via `...` menu in navigation bar)
6. Verify Add Account still works (now access via `...` menu in navigation bar while on Accounts tab)

## Screenshots:

Accounts - light | Accounts - dark
--|--

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

**Comment by nuo-xu** [comment]
we probably want to add the ability to configure the space value and how they lie on each other(`.zIndex`) for `MultipleCircleIconView`
also the design for `MultipleCircleIconView` when icons are more than `maxIcons` is actually the `+x` instead of `...` as in  multiple network icons. 
Screenshot 2024-01-18 at 11 55 24 AM
vs
Screenshot 2024-01-18 at 11 55 39 AM

and we probably want to give a background color for `AssetIconView` since some icon image has transparent background
for example below shows the `Raydium` icon has transparent background
Screenshot 2024-01-18 at 11 59 29 AM

File: Sources/BraveWallet/Crypto/Stores/AccountsStore.swift [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+      walletService: walletService,
+      _onNetworkListChanged: { [weak self] in
+        self?.update()
+      }
+    )
+    Preferences.Wallet.showTestNetworks.observe(from: self)
+  }
+  
+  func tearDown() {
+    keyringServiceObserver = nil
```
Comment (nuo-xu) [inline Sources/BraveWallet/Crypto/Stores/AccountsStore.swift:87]: need to nil `walletServiceObserver` too

Comment (StephenHeaps) [reply Sources/BraveWallet/Crypto/Stores/AccountsStore.swift:87]: Nice catch, fixed in https://github.com/brave/brave-ios/pull/8659/commits/6824b33977f8cd4806000e379946f52bb16ddee6

**Comment by StephenHeaps** [comment]
> we probably want to add the ability to configure the space value and how they lie on each other(`.zIndex`) for `MultipleCircleIconView` also the design for `MultipleCircleIconView` when icons are more than `maxIcons` is actually the `+x` instead of `...` as in multiple network icons. 

Discussed offline; these tokens are sorted by balance so we chose to keep z-index order so largest balance remains on top. Investigated the `+x`, but no perfect solution with larger numbers and the `+` would likely overlapped without larger spacing increase between icons.
 
> and we probably want to give a background color for `AssetIconView` since some icon image has transparent background for example below shows the `Raydium` icon has transparent background 

Good idea. Updated so we have a circular background instead of just a circular stroke 👍. https://github.com/brave/brave-ios/pull/8659/commits/0b43005cccc5b064e88293e1bd07b9be3b44a117

**Review by nuo-xu — APPROVED** [review]

**Pull request merged**